src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -341,10 +341,13 @@ private BoundStatement RewriteUsingStatementTryFinally(SyntaxNode syntax, BoundB
                     hasErrors: false);
             }
 
+            // Terminate the try block with a hidden sequence point
+            BoundBlock terminatedTryBlock = tryBlock.WithStatements(tryBlock.Statements.Add(new BoundSequencePoint(null, null)));
```

Comment:
Yes, I wonder about that.

I can find the same problem with `while` and `for` loops:
```C#
    using (var stream = new System.IO.MemoryStream())
            while (x)
                x = false;
```
```C#
   using (var stream = new System.IO.MemoryStream())
            for ( ; x == true; )
                x = false;
```

On the other hand, I suspect there are other scenarios that introduce a `try/catch`. So far, we fixed `async` and `using`, but I think there are more. I'll try to confirm if more scenarios are broken.